### ISSUE TITLE
Release plan for the next major release v0.2.0-alpha

### ISSUE THREAD

**Issue by imdeepmind** [imdeepmind/NeuralPy#38]
# Release plan for the next major release v0.2.0-alpha
## New features/fixes needed to be added:
  - [x] Layers
    - [x] Support for Conv1D, Conv2D, Conv3D
    - [x] Support for Flatten layer
    - [x] Support for MaxPool1D, MaxPool2D, MaxPool3D
    - [x] Support for AvgPool1D, AvgPool2D, AvgPool3D
    - [x] Support for RNN,  GRU, LSTM
    - [x] Support for RNNCell,  GRUCell, LSTMCell
    - [x] Support for BatchNorm1D, BatchNorm2D, BatchNorm3D
    - [x] Support for Embedding layer
  - [x] Activation Functions
    - [x] Support for GELU
    - [x] Support for SELU
  - [x] Regularizers
    - [x] Dropout2D
    - [x] Dropout3D
    - [x] AlphaDropout
  - [x] Generator Support
    - [x] Generator based `fit` method
    - [x] Generator based `predict` and `predict_class` method
    - [x] Generator based `evaluate` method
  - [ ] Other
    - [ ] Bug Fix
    - [x] More Stability
    - [x] Better documentation
    - [x] More Test Scripts to achieve 90% test coverage
    - [x] Improving code quality (pylint score of >= 9.5)
    - [x] Ability to save and load models
    - [x] Training Logger

**Comment by imdeepmind** [comment]
@imdeepmind to work on Conv layers, and generator support

**Comment by jeffin07** [comment]
Working on RNN layers

**Comment by imdeepmind** [comment]
achieved 76% test coverage, so increasing the test coverage target to 90%

**Comment by imdeepmind** [comment]
@jeffin07 can you work on Embedding layer and Batch Norm?

**Comment by jeffin07** [comment]
@imdeepmind sure.Currently working on Embedding layer

**Comment by imdeepmind** [comment]
Delayed the release date to 2nd August so that we can use the weekend

**Comment by imdeepmind** [comment]
Remaining work: Testing RNN, LSTM, GRU, RNNCell, LSTMCell, GRUCell layers and fix bugs
